server/src/__tests__/executables.test.ts

```diff
@@ -0,0 +1,44 @@
+import * as path from 'path'
+import { Executables } from '../executables'
+
+const executables = Executables.fromPath(
+  path.resolve(__dirname, '..', '..', '..', 'testing', 'executables'),
+)
+
+describe('list', () => {
+  it('It finds executables on the PATH', () => {
+    expect.assertions(1)
+    const result = executables.then(ex => ex.list().find(x => x === 'iam-executable'))
+    return expect(result).resolves.toBeTruthy()
+  })
+
+  it.skip('It only considers files that have the executable bit set', () => {
+    expect.assertions(1)
+    const result = executables.then(ex => ex.list().find(x => x === 'iam-not-executable'))
+    return expect(result).resolves.toBeFalsy()
+  })
+
+  it('It only considers executable directly on the PATH', () => {
+    expect.assertions(1)
+    const result = executables.then(ex =>
```

Comment:
Using `await` in these tests might make it easier. Then you don’t need the expect.assertions. : )